iothub/device/src/Message.cs

```diff
@@ -57,374 +36,36 @@ public Message(Stream stream)
-            set => SystemProperties[MessageSystemPropertyNames.UserId] = value;
-        }
-
-        /// <summary>
-        /// For outgoing messages, contains the Mqtt topic that the message is being sent to
-        /// For incoming messages, contains the Mqtt topic that the message arrived on
-        /// </summary>
-        internal string MqttTopicName { get; set; }
-
-        /// <summary>
-        /// Used to specify the schema of the message content.
-        /// </summary>
-        public string MessageSchema
+            : base(stream, streamDisposalResponsibility)
         {
-            get => GetSystemProperty<string>(MessageSystemPropertyNames.MessageSchema);
-            set => SystemProperties[MessageSystemPropertyNames.MessageSchema] = value;
         }
 
-        /// <summary>
-        /// Custom date property set by the originator of the message.
-        /// </summary>
-        public DateTime CreationTimeUtc
-        {
-            get => GetSystemProperty<DateTime>(MessageSystemPropertyNames.CreationTimeUtc);
-            set => SystemProperties[MessageSystemPropertyNames.CreationTimeUtc] = value;
-        }
-
-        /// <summary>
-        /// True if the message is set as a security message
-        /// </summary>
-        public bool IsSecurityMessage => CommonConstants.SecurityMessageInterfaceId.Equals(GetSystemProperty<string>(MessageSystemPropertyNames.InterfaceId), StringComparison.Ordinal);
-
         /// <summary>
         /// Used to specify the content type of the message.
         /// </summary>
         public string ContentType
         {
-            get => GetSystemProperty<string>(MessageSystemPropertyNames.ContentType);
-            set => SystemProperties[MessageSystemPropertyNames.ContentType] = value;
-        }
-
-        /// <summary>
-        /// Specifies the input name on which the message was sent, if there was one.
-        /// </summary>
-        public string InputName
-        {
-            get => GetSystemProperty<string>(MessageSystemPropertyNames.InputName);
-            internal set => SystemProperties[MessageSystemPropertyNames.InputName] = value;
-        }
-
-        /// <summary>
-        /// Specifies the device Id from which this message was sent, if there is one.
-        /// </summary>
-        public string ConnectionDeviceId
-        {
-            get => GetSystemProperty<string>(MessageSystemPropertyNames.ConnectionDeviceId);
-            internal set => SystemProperties[MessageSystemPropertyNames.ConnectionDeviceId] = value;
-        }
-
-        /// <summary>
-        /// Specifies the module Id from which this message was sent, if there is one.
-        /// </summary>
-        public string ConnectionModuleId
-        {
-            get => GetSystemProperty<string>(MessageSystemPropertyNames.ConnectionModuleId);
-            internal set => SystemProperties[MessageSystemPropertyNames.ConnectionModuleId] = value;
+            get => PayloadContentType;
```

Comment:
the publicly visible field on `Message` is called `ContentType`, which the internal backing field on `MessageBase` is called `PayloadContentType`.